test/lib/shellcheck-services.nix

```diff
@@ -45,18 +45,17 @@ let
     # Return set of services ({ service1 = true; service2 = true; ... })
     # which are either defined or not defined within `sourcePrefix`, depending
     # on `shouldMatch`.
-    collectServices = shouldMatch: lib.listToAttrs (builtins.concatLists (zipListsWith (services: file:
+    collectServices = shouldMatch: lib.listToAttrs (builtins.concatLists (map (def:
       let
+        services = def.value;
+        inherit (def) file;
         isMatching = lib.hasPrefix sourcePrefix file;
       in
         # Nix has no boolean XOR, so use `if`
         lib.optionals (if shouldMatch then isMatching else !isMatching) (
           (map (service: { name = service; value = true; }) (builtins.attrNames services))
         )
-    # TODO-EXTERNAL:
```

Comment:
After [my fix](https://github.com/NixOS/nixpkgs/pull/133622) it broke again, and I've ignored it since then when checking `TODO-EXTERNAL`s.
But, indeed, it's working now.
This this by running a nested  `doas` cmd (`lncli` internally calls doas):
```bash
run-tests.sh -s '{
  imports = [ (nix-bitcoin + "/modules/presets/secure-node.nix" )];
  services.lnd.enable = true;
}' container --run c doas -u operator lncli getinfo

```